src/Toml/Parser/TOML.hs

```diff
@@ -1,44 +1,47 @@
 module Toml.Parser.TOML
        ( keyValP
        , tableHeaderP
+       , inlineTableP
        , tomlP
        ) where
 
-import Control.Applicative (Alternative (many))
+import Control.Applicative (Alternative (many, (<|>)))
+import Control.Applicative.Combinators (sepEndBy, between)
+import Data.Either (lefts, rights)
 
-import Toml.Parser.Core (Parser, sc, text)
-import Toml.Parser.Value (keyP, tableNameP, valueP)
+import Toml.Parser.Core (Parser, sc, text, char, try)
+import Toml.Parser.Value (keyP, tableNameP, anyValueP)
 import Toml.PrefixTree (Key (..), fromList)
-import Toml.Type (AnyValue, TOML (..), typeCheck)
+import Toml.Type (AnyValue, TOML (..))
 
 import qualified Data.HashMap.Lazy as HashMap
 
+hasKeyP :: Parser (Either (Key, AnyValue) (Key, TOML))
+hasKeyP = try (Left <$> keyValP) <|> (Right <$> inlineTableP)
```

Comment:
I think that this parser should be rewrittern in the following way:

1. Change type to `Parser (Key, Either TOML AnyValue)`. You will always parse key, so it makes sense to move it out.
2. If you parse `inlineTable` first than you don't need `try` and I think that this is better.
3. You can use `eitherP` from `parser-combinators` library:
    * https://hackage.haskell.org/package/parser-combinators-1.0.0/docs/Control-Applicative-Combinators.html#v:eitherP